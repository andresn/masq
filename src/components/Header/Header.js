import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { Avatar } from 'components'
import { Chevron, Home } from 'icons'

import { UserContext } from 'context/user'

import './Header.css'

function TitleSection (props) {
  return (
    <div id='title-section'>
      <p id='title'>Masq</p>
      <p>&nbsp;by Qwant</p>
    </div>
  )
}

function AvatarSection (props) {
  const { image, username } = props
  return (
    <div id='avatar-section'>
      <Avatar image={image} />
      <span id='avatar-text' style={{fontSize: '14px'}}>{username}</span>
      <Chevron className='caret' />
    </div>
  )
}

export default function Header (props) {
  const { shadow, children, onLogout, childrenHeight } = props
  const height = childrenHeight ? 80 + childrenHeight : 64

  return (
    <UserContext.Consumer>
      {user =>
        <div>
          <div id='invisible' style={{height: height}} />
          <div className={'Header' + (shadow ? ' shadow' : '')}>
            <div id='top-section'>
              {user ? <AvatarSection username={user.username} image={user.image} /> : <TitleSection /> }
              <Link to='login'>
                <Home className='home' onClick={onLogout} />
              </Link>
            </div>
            {children}
            {children ? (<div className='children' />) : null}
          </div>
        </div>
      }
    </UserContext.Consumer>
  )
}

Header.propTypes = {
  shadow: PropTypes.bool,
  onLogout: PropTypes.func,
  children: PropTypes.element,
  childrenHeight: PropTypes.number
}
