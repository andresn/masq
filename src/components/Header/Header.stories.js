import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router'

import { UserContext } from 'context/user'

import Header from './Header'

const user = {
  username: 'some username',
  firstname: 'some firstname',
  lastname: 'some lastname',
  image: 'https://randomuser.me/api/portraits/women/79.jpg'
}

storiesOf('Header', module)
  .add('default header', () => (
    <MemoryRouter><Header /></MemoryRouter>
  ))
  .add('with shadow', () => (
    <MemoryRouter><Header shadow /></MemoryRouter>
  ))
  .add('with a user logged in', () => (
    <MemoryRouter>
      <UserContext.Provider value={user}>
        <Header />
      </UserContext.Provider>
    </MemoryRouter>
  ))
