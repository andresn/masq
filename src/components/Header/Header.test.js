import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import { UserContext } from 'context/user'

import Header from './Header'

it('renders Header', () => {
  const component = renderer.create(<MemoryRouter><Header /></MemoryRouter>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with shadow', () => {
  const component = renderer.create(<MemoryRouter><Header shadow /></MemoryRouter>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with a user', () => {
  const user = {
    username: 'some username',
    firstname: 'some firstname',
    lastname: 'some lastname',
    image: 'https://randomuser.me/api/portraits/women/79.jpg'
  }
  const component = renderer.create(
    <MemoryRouter>
      <UserContext.Provider value={user}>
        <Header />
      </UserContext.Provider>
    </MemoryRouter>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
