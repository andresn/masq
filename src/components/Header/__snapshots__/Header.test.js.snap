// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header 1`] = `
<div>
  <div
    id="invisible"
    style={
      Object {
        "height": 64,
      }
    }
  />
  <div
    className="Header"
  >
    <div
      id="top-section"
    >
      <div
        id="title-section"
      >
        <p
          id="title"
        >
          Masq
        </p>
        <p>
           by Qwant
        </p>
      </div>
      <a
        href="login"
        onClick={[Function]}
      >
        <svg
          className="home"
          height={22}
          onClick={undefined}
          viewBox="0 0 20 22"
          width={20}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M19.6 7.217l-9-6.992c-.4-.3-.9-.3-1.2 0l-9 6.992c-.3.2-.4.5-.4.799v10.987C0 20.701 1.3 22 3 22h14c1.7 0 3-1.299 3-2.997V8.016c0-.3-.1-.6-.4-.8zM12 20.002H8v-7.99h4v7.99zm6-.999c0 .6-.4 1-1 1h-3v-8.99c0-.6-.4-1-1-1H7c-.6 0-1 .4-1 1v8.99H3c-.6 0-1-.4-1-1V8.515l8-6.193 8 6.193v10.488z"
              id="home-svg"
            />
          </defs>
          <use
            fill="currentColor"
            fillRule="nonzero"
            xlinkHref="#home-svg"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders Header with a user 1`] = `
<div>
  <div
    id="invisible"
    style={
      Object {
        "height": 64,
      }
    }
  />
  <div
    className="Header"
  >
    <div
      id="top-section"
    >
      <div
        id="title-section"
      >
        <p
          id="title"
        >
          Masq
        </p>
        <p>
           by Qwant
        </p>
      </div>
      <a
        href="login"
        onClick={[Function]}
      >
        <svg
          className="home"
          height={22}
          onClick={undefined}
          viewBox="0 0 20 22"
          width={20}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M19.6 7.217l-9-6.992c-.4-.3-.9-.3-1.2 0l-9 6.992c-.3.2-.4.5-.4.799v10.987C0 20.701 1.3 22 3 22h14c1.7 0 3-1.299 3-2.997V8.016c0-.3-.1-.6-.4-.8zM12 20.002H8v-7.99h4v7.99zm6-.999c0 .6-.4 1-1 1h-3v-8.99c0-.6-.4-1-1-1H7c-.6 0-1 .4-1 1v8.99H3c-.6 0-1-.4-1-1V8.515l8-6.193 8 6.193v10.488z"
              id="home-svg"
            />
          </defs>
          <use
            fill="currentColor"
            fillRule="nonzero"
            xlinkHref="#home-svg"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders Header with shadow 1`] = `
<div>
  <div
    id="invisible"
    style={
      Object {
        "height": 64,
      }
    }
  />
  <div
    className="Header shadow"
  >
    <div
      id="top-section"
    >
      <div
        id="title-section"
      >
        <p
          id="title"
        >
          Masq
        </p>
        <p>
           by Qwant
        </p>
      </div>
      <a
        href="login"
        onClick={[Function]}
      >
        <svg
          className="home"
          height={22}
          onClick={undefined}
          viewBox="0 0 20 22"
          width={20}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M19.6 7.217l-9-6.992c-.4-.3-.9-.3-1.2 0l-9 6.992c-.3.2-.4.5-.4.799v10.987C0 20.701 1.3 22 3 22h14c1.7 0 3-1.299 3-2.997V8.016c0-.3-.1-.6-.4-.8zM12 20.002H8v-7.99h4v7.99zm6-.999c0 .6-.4 1-1 1h-3v-8.99c0-.6-.4-1-1-1H7c-.6 0-1 .4-1 1v8.99H3c-.6 0-1-.4-1-1V8.515l8-6.193 8 6.193v10.488z"
              id="home-svg"
            />
          </defs>
          <use
            fill="currentColor"
            fillRule="nonzero"
            xlinkHref="#home-svg"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
